core/src/main/java/io/questdb/cairo/TableUtils.java

```diff
@@ -1044,26 +1033,27 @@ public static int lengthOf(@Nullable CharSequence columnValue) {
     }
 
     public static @NotNull MatViewDefinition loadMatViewDefinition(
-            FilesFacade ff,
-            MemoryCMR mem,
+            MetaFileReader reader,
             Path path,
             int rootLen,
             TableToken matViewToken
     ) {
-        path.trimTo(rootLen)
-                .concat(matViewToken.getDirName())
-                .concat(MAT_VIEW_FILE_NAME);
-        mem.smallFile(ff, path.$(), MemoryTag.MMAP_DEFAULT);
-        if (mem.size() < MV_HEADER_SIZE + 3 * Long.BYTES + 3 * Integer.BYTES + Character.BYTES) {
+        path.trimTo(rootLen).concat(matViewToken.getDirName()).concat(MAT_VIEW_FILE_NAME);
+        reader.of(path.$());
+        MetaFileReader.BlockCursor cursor = reader.getCursor();
+        if (cursor.hasNext()) {
+            return loadMatViewDefinitionMessage(cursor.next(), matViewToken);
+        } else {
             throw CairoException.critical(0)
-                    .put("cannot read materialized view definition, file is too small [path=")
+                    .put("cannot read materialized view definition, file is empty [path=")
                     .put(path)
                     .put(']');
         }
+    }
 
-        long offset = MV_HEADER_SIZE;
-
-        final CharSequence baseTableName = mem.getStrA(offset);
+    private static MatViewDefinition loadMatViewDefinitionMessage(final ReadableBlock mem, final TableToken matViewToken) {
```

Comment:
For some reason I thought that every field here would go into its own block. If that's not the case and all initial fields go together, we need to at least validate block type - otherwise we won't be able to cope with future format versions.